File: Competetive programming/Python/Unit 5/Primary_Arithmetic.py
print("===== PRIMARY ARITHMETIC =====\n")

num1 = int(input("Enter First Number: "))
num2 = int(input("Enter Second Number: "))

carry = 0
carry_count = 0

while num1 > 0 or num2 > 0:

    digit1 = num1 % 10
    digit2 = num2 % 10

    total = digit1 + digit2 + carry

    if total >= 10:
        carry = 1
        carry_count += 1
    else:
        carry = 0

    num1 //= 10
    num2 //= 10

if carry_count == 0:
    print("\nNo carry operation.")
elif carry_count == 1:
    print("\n1 carry operation.")
else:
    print(f"\n{carry_count} carry operations.")

File: Competetive programming/C++/Unit 5/Primary_Arithmetic.cpp
#include <iostream>
using namespace std;
int main() {
    long long num1, num2;
    cout << "===== PRIMARY ARITHMETIC =====\n\n";
    cout << "Enter First Number:\n";
    cout << "Example: 123\n";
    cin >> num1;
    cout << "\nEnter Second Number:\n";
    cout << "Example: 594\n";
    cin >> num2;
    int carry = 0;
    int carryCount = 0;
    while(num1 > 0 || num2 > 0) {
        int digit1 = num1 % 10;
        int digit2 = num2 % 10;
        int sum = digit1 + digit2 + carry;
        if(sum >= 10) {
            carry = 1;
            carryCount++;
        }
        else {
            carry = 0;
        }
        num1 /= 10;
        num2 /= 10;
    }
    if(carryCount == 0)
        cout << "\nNo carry operation.\n";
    else if(carryCount == 1)
        cout << "\n1 carry operation.\n";
    else
        cout << "\n" << carryCount << " carry operations.\n";
    return 0;
}